### DISCUSSION TITLE
Passing electron-log from main to preload

### DISCUSSION THREAD

**Discussion by moltenice** [megahertz/electron-log#298] [Q&A]
Given:

```javascript
webPreferences: {
  nodeIntegration: false,
  contextIsolation: true,
  sandbox: true,
}
```

electron-log is not available by default in the preload script if I understand correctly. Trying to use ipcRenderer to pass through the log functions gives: Error: An object could not be cloned.

How do I use electron-log here?

**Comment by megahertz** [comment]
I have plans to make it easier to use the library in such an envitonment. Currently, you can use an approach [like that ](https://github.com/megahertz/electron-log/blob/master/e2e/isolation/preload.js).
